test/e2e/dra/test-driver/app/controller.go

```diff
@@ -30,24 +30,101 @@ import (
 	v1 "k8s.io/api/core/v1"
 	resourcev1alpha2 "k8s.io/api/resource/v1alpha2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
+	listersv1 "k8s.io/client-go/listers/core/v1"
 	"k8s.io/dynamic-resource-allocation/controller"
 	"k8s.io/klog/v2"
 )
 
 type Resources struct {
+	DriverName         string
 	DontSetReservedFor bool
 	NodeLocal          bool
-	Nodes              []string
-	MaxAllocations     int
-	Shareable          bool
+	// Nodes is a fixed list of node names on which resources are
+	// available. Mutually exclusive with NodeLabels.
+	Nodes []string
```

Comment:
Question: Since `Nodes` and `NodeLabels` are mutually exclusive, should we not make these members private so as to prevent inadvertent access?

Reply:
If they were private, then `json.Unmarshal` would not be able to set them. This struct is basically just a bunch of values.